azurerm/resources.go

```diff
@@ -0,0 +1,135 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/cycloidio/terracognita/provider"
+	"github.com/cycloidio/terracognita/tag"
+)
+
+// ResourceType is the type used to define all the Resources
+// from the Provider
+type ResourceType int
+
+//go:generate enumer -type ResourceType -addprefix azurerm_ -transform snake -linecomment
+const (
+	ResourceGroup          ResourceType = iota
+	VirtualMachine         ResourceType = iota
+	VirtualNetwork         ResourceType = iota
+	Subnet                 ResourceType = iota
+	NetworkInterface       ResourceType = iota
+	NetworkSecurityGroup   ResourceType = iota
+	VirtualMachineScaleSet ResourceType = iota
+)
+
+type rtFn func(ctx context.Context, a *azurerm, resourceType string, tags []tag.Tag) ([]provider.Resource, error)
+
+var (
+	resources = map[ResourceType]rtFn{
+		ResourceGroup:          resourceGroup,
+		VirtualMachine:         virtualMachines,
+		VirtualNetwork:         cacheVirtualNetworks,
+		Subnet:                 subnets,
+		NetworkInterface:       networkInterfaces,
+		NetworkSecurityGroup:   networkSecurityGroups,
+		VirtualMachineScaleSet: virtualMachineScaleSets,
+	}
+)
+
+func resourceGroup(ctx context.Context, a *azurerm, resourceType string, tags []tag.Tag) ([]provider.Resource, error) {
+	resourceGroup := a.azurer.GetResourceGroup()
+	resources := make([]provider.Resource, 0)
```

Comment:
i suppose there is only 1 resource group ? In this case, we could set the length of the array
```suggestion
	resources := make([]provider.Resource, 0, 1)
```

or you could try to directly create the slice with the provider.Resource inside.